File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/factory_method/MulFactory.java
package jv.com.tencent.cain.design_pattern.factory_method;

/**
 * @author cainjiang
 * @date 2019-05-20
 */
public class MulFactory implements IOperationFactory {
    @Override
    public Operation createOpration() {
        MulOperation mulOperation = new MulOperation();
        mulOperation.operationSymbol = "*";
        mulOperation.operationName = "乘法";
        return mulOperation;
    }

}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/factory_method/MulFactory.kt
package kt.com.tencent.cain.desigin_pattern.factory_method

import kt.com.tencent.cain.desigin_pattern.simple_factoty.MulOperation
import kt.com.tencent.cain.desigin_pattern.simple_factoty.Operation

/**
 * @author cainjiang
 * @date 2019-05-21
 */
class MulFactory : IOperationFactory {
    override fun createOpration(): Operation {
        val mulOperation = MulOperation()
        mulOperation.operationSymbol = "*"
        mulOperation.operationName = "乘法"
        return mulOperation
    }
}